### PR TITLE
Add idol-library ACI Server templates

### PR DESCRIPTION
**Pull request by daniel-lau-mf**
This encourages us to make sure all basic ACI servers use the same value names for consistency and
support common features

Implemented idol-view chart using these templates
(wants a separate submit to update the idol-library dependency to a published version).

### CODE REVIEW

**Comment by daniel-lau-mf** [comment]
Note that in most of our existing charts we have `enableIngress: true` but I have renamed to `ingressEnabled: true` to attempt to group the ingress parameters.

Wondered if there was a better way to group idolImageRegistry, idolVersion, imagePullSecrets and image similarly.

**Comment by christopher-rayson-mf** [comment]
> Wondered if there was a better way to group idolImageRegistry, idolVersion, imagePullSecrets and image similarly.

Would any of these be candidates for global values?

File: helm/idol-library/templates/aciserver/_service.tpl [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Values.name | quote }}
+  labels: {{- include "idol-library.labels" . | nindent 4 }}
+spec:
+  ports:
+  - port: {{ .Values.aciPort | int }}
+    targetPort: aci-port
+    name: aci-port
```
Comment (christopher-rayson-mf) [inline helm/idol-library/templates/aciserver/_service.tpl:24]: Worth including an optional index port block here? (Guarded with `{{- if .Values.indexPort }}` or similar.)

Comment (christopher-rayson-mf) [reply helm/idol-library/templates/aciserver/_service.tpl:24]: Guess that'd need a corresponding block in the deployment template.

Comment (daniel-lau-mf) [reply helm/idol-library/templates/aciserver/_service.tpl:24]: probably only want an index port for statefulsets?

Comment (daniel-lau-mf) [reply helm/idol-library/templates/aciserver/_service.tpl:24]: though addiing to the service definition would be fine

Comment (christopher-rayson-mf) [reply helm/idol-library/templates/aciserver/_service.tpl:24]: > probably only want an index port for statefulsets?

Fair point.

> though addiing to the service definition would be fine

App selector in the service would work with either a deployment or a stateful set, right?

**Comment by daniel-lau-mf** [comment]
> Finding it slightly unintuitive that the global values override the local ones rather than the other way around (my instinct is that local definitions are "more specific" so should take precedence). Is that to avoid setting globals in every subchart?

Mostly that, yes. I don't really know how Helm will behave with multiple charts defining defaults for the same global value, so think we should try to avoid it.

**Pull request merged**